### ISSUE TITLE
Nothing received from server on UDP port 60001

### ISSUE THREAD

**Issue by callumacrae** [mobile-shell/mosh#489]
I think this is basically a duplicate of #374, but that's been fixed.

I can connect to one of my servers fine, but the server that is on AWS won't let me connect to: I just get the error in the title.

Running `netstat -nulp | grep mosh` reveals that it is bound to the wrong IP (it's bound to the internal IP, not the actual IP), and I guess that's what is causing the problem. I don't know how to fix it, however.

The port is unblocked.

**Comment by 2i** [comment]
In my case there was an issue with the configuration of the ssh service in systemd. As written in http://unix.stackexchange.com/a/176292/147881, when using ssh.socket or not using `KillMode=process`, ending the ssh connection kills all child processes.

This also affects the mosh-server process. So when mosh connects to the server, it successfully starts `mosh-server` and closes the ssh connection. But when it tries to connect to the server via UDP afterwards the `mosh-server` process has been killed. The solution is to enable and start `ssh.service` and disable and stop `ssh.socket` systemctl. It is also required to restart the ssh connection. Maybe also the `KillMode=process` muss be added in `/etc/systemd/system/sshd.service` under `[Sevice]`.

**Comment by cgull** [comment]
@2i, your issue is a duplicate of #399.  As I note there, `pam_systemd` appears to be relevant.

**Comment by reuben** [comment]
For others running into this issue, a way I found for working around this problem without changing the sshd configuration is by using `--server="systemd-run --scope --user mosh-server"`.
